src/org/labkey/test/components/ui/entities/EntityInsertPanel.java

```diff
@@ -102,73 +108,105 @@ public EntityInsertPanel addRecords(List<Map<String, Object>> records)
 
     public EntityInsertPanel setRecordValues(Map<String, Object> columnValues)
     {
-        int insertRowIndex = elementCache().grid.getEditableRowIndices().get(0);
+        int insertRowIndex = grid().getEditableRowIndices().get(0);
         return setRecordValues(columnValues, insertRowIndex);
     }
 
     public EntityInsertPanel setRecordValues(Map<String, Object> columnValues, int row)
     {
         for(String columnName : columnValues.keySet())
         {
-            elementCache().grid.setCellValue(row, columnName, columnValues.get(columnName));
+            grid().setCellValue(row, columnName, columnValues.get(columnName));
         }
 
         return this;
     }
 
     public EditableGrid getEditableGrid()
     {
-        return elementCache().grid;
+        showGrid();
+        return grid();
+    }
+
+    public FileUploadPanel getFileUploadPanel()
+    {
+        showFileUpload();
+        return fileUploadPanel();
+    }
```

Comment:
If I understand what component this is address, this is incorrect. The upload panel is not a sub panel in the insert panel, it is a separate panel on the page. This should not be nested here. It would be helpful if you provided a screenshot, or some other reference, in your PR description to clarify.

SampleManager has these as two different pages (similar to how other automation in LabKey is done). SampleImportPage covers this functionality.

Reply:
I've included a screen-grab of the file-upload panel part of the EntityUploadPanel from a non-biologics/non-SampleManager view, to illustrate that file upload is a selectable mode of the component, the panel is there in it

Or, we could just not do this and only support grid functions in this component, it's not a hill I need to die on here
I only added it here because I wanted a functional waitForLoaded() function that wouldn't be broken if the page for some reason opened the component in file upload mode

Reply:
A quick look through EntityInsertPanel.tsx   [labkey-ui-components/packages/components/src/internal/components/entities/EntityInsertPanel.tsx]
tells me that file upload/add/remove/submit are parts of what it does.